src/lib/core-extension-service.ts

```diff
@@ -139,5 +163,27 @@ interface CoreTelemetryService {
 interface CoreExtensionApi {
 	services: {
 		TelemetryService: CoreTelemetryService;
+		WorkspaceContext: WorkspaceContext;
 	}
 }
+
+interface WorkspaceContext {
+	readonly onOrgChange: Event<{
+	username?: string;
+	alias?: string;
+	}>;
+	getInstance(forceNew: boolean): WorkspaceContext;
+	getConnection(): Promise<Connection>;
+	username(): string | undefined;
+	alias(): string | undefined;
+}
+
+interface Connection {
+	instanceUrl: string;
+	getApiVersion(): string;
+	getUsername(): string | undefined;
+	getAuthInfoFields(): AuthFields;
+	request<T>(options: { method: string; url: string; body: string; headers?: Record<string, string> }): Promise<T>;
+}
```

Comment:
I understand that we need to define these because we are pulling the CoreServiceApi from line 35. But I wonder... is it frowned upon to have a dependency on that module so we can pull in its interfaces instead of having to reimplement them here.

This pattern just seems like a flaw in the design of how we are getting access to the CoreExtensionApi's services.

Pulling in all those AuthFields for example... it just feels like at some point we are going to get out of sync with the offical types and interfaces that live in that module.

Reply:
Agree with your concerns. I didn't find the types exported from the core and e4d followed this pattern. This is already a code smell because now we have these types and interfaces defined at two places. AFAIK, types cannot be exported from one extension, just APIs. So the ideal solution would be to bundle these all into a NPM package that both the core and all the consumer extensions depend on. I will initiate a talk with the extension folks on this. We will have to live with this until then, unfortunately.

Reply:
Yes we should definitely talk with them.

Ideally... they wouldn't require their extension to be installed... but instead allow us to mark a module as a dependency to get these services.